src/rgw/rgw_lua_data_filter.cc

```diff
@@ -84,33 +84,37 @@ struct BufferlistMetaTable : public EmptyMetaTable {
 };
 
 int RGWObjFilter::execute(bufferlist& bl, off_t offset, const char* op_name) const {
-  auto L = luaL_newstate();
-  lua_state_guard lguard(L);
+  lua_state_guard lguard(s->cct->_conf->rgw_lua_max_memory_per_state);
+  auto L = lguard.get();
+  if (!L) {
+    ldpp_dout(s, 1) << "Failed to create state for Lua data context" << dendl;
+    return -ENOMEM;
+  }
 
-  open_standard_libs(L);
+  try {
+    open_standard_libs(L);
 
-  create_debug_action(L, s->cct);  
+    create_debug_action(L, s->cct);  
 
-  // create the "Data" table
-  create_metatable<BufferlistMetaTable>(L, true, &bl);
-  lua_getglobal(L, BufferlistMetaTable::TableName().c_str());
-  ceph_assert(lua_istable(L, -1));
+    // create the "Data" table
+    create_metatable<BufferlistMetaTable>(L, true, &bl);
```

Comment:
what is a BufferlistMetaTable?  why is it needed?

Reply:
it is used for the "data" context where you can access the payload of the request